examples/ecs/generic_system.rs

```diff
@@ -0,0 +1,91 @@
+use bevy::{ecs::component::Component, prelude::*};
+
+/// Generic types allow us to reuse logic across many related systems,
+/// allowing us to specialize our function's behavior based on which type (or types) are passed in.
+///
+/// This is commonly useful for working on related components or resources,
+/// where we want to have unique types for querying purposes but want them all to work the same way.
+/// This is particularly powerful when combined with user-defined traits to add more functionality to these related types.
+/// Remember to insert a specialized copy of the system into the schedule for each type that you want to operate on!
+///
+/// For more advice on working with generic types in Rust, check out https://doc.rust-lang.org/book/ch10-01-syntax.html
+/// or https://doc.rust-lang.org/rust-by-example/generics.html
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+enum AppState {
+    MainMenu,
+    InGame,
+}
+
+#[derive(Component)]
+struct TextToPrint(String);
+
+#[derive(Component)]
+struct Timer(bevy::prelude::Timer);
```

Comment:
Hey @alice-i-cecile I couldn't find in the 0.5 -> 0.6 docs how this is meant to be wrapped now but I saw you post this in discord so I've taken it on here. Please let me know the "correct" way to do this if this isn't it.

Reply:
This change happened after the 0.6, a few days ago in #2982
It would be better to use a meaningful name for the wrapper... `PrinterTick` maybe?